install.sh

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+waldo_cli_bin="${WALDO_CLI_BIN:-}"
+waldo_cli_url="${WALDO_CLI_URL:-https://github.com/waldoapp/waldo-go-cli/releases/latest/download}"
+
+waldo_exec1_name="waldo"
+waldo_exec2_name="sim_appcenter_build_and_upload.sh"
+
+waldo_asset1_name=
+waldo_asset2_name=
+
+waldo_install_detected=false
+waldo_macos_detected=false
+
+waldo_bash_profile="${HOME}/.bash_profile"
+waldo_bashrc="${HOME}/.bashrc"
+waldo_zshrc="${ZDOTDIR:-${HOME}}/.zshrc"
+
+function check_curl_command() {
+    if ! command -v curl >/dev/null; then
+        fail "No ‘curl’ command found"
+    fi
+}
+
+function check_destination() {
+    local _cli_path=$(command -v waldo)
+    local _cli_bin=
+
+    if [[ -n $_cli_path &&  -e $_cli_path  ]]; then
+        waldo_cli_bin=$(dirname $_cli_path)
+
+        echo "Waldo CLI installation detected in ‘${waldo_cli_bin}’ -- will re-install"
+        echo ""
+
+        waldo_install_detected=true
+    else
+        if [[ -z $waldo_cli_bin ]]; then
+			waldo_cli_bin="${HOME}/.waldo/bin"
+        fi
+
+        echo "Waldo CLI will be installed in ‘${waldo_cli_bin}’"
+        echo ""
+    fi
+}
+
+function check_installation() {
+    if [[ $waldo_install_detected == true ]]; then
+        echo ""
+        echo "Waldo CLI successfully re-installed!"
+    else
+		if [[ $waldo_macos_detected == true ]]; then
```

Comment:
Is it me or are you going to create a `.bash_profile` and a `.zshrc` no matter what (if they don't exist) ?

Reply:
Maybe I can't read bash correctly but it looks like you are going to echo `Updating your PATH...` twice no matter what (and modify 2 files).

I would just check if one exists and if it does not just print "add to your path". Brew does something like that with `caveats` I believe ?

Reply:
I basically adapted what Maestro does in their installer script. It is not clear (to me) whether there is any _definitive_ way to tell which shell (bash, zsh, or something else) is currently using (or will use in the future). Thus, the sledgehammer of adding/modifying 2 files. Also, it is _very_ easy to end up appending to one of those files unnecessarily. Which is why I personally despise installer scripts that do so. I may have misunderstood Laurent’s intent, but it seems to me he is in favor of “the Maestro way”.

Reply:
@laurentsigal 
Perhaps you would care to weigh in on this before I merge this and it becomes public.

Reply:
Yes, I would edit those files when they exist, but not generate them otherwise.